src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/adjust_brgemm_copy_b_loop_ports.cpp

```diff
@@ -65,17 +65,27 @@ bool pass::AdjustBrgemmCopyBLoopPorts::run(const snippets::lowered::LinearIR& li
 
     bool modified = false;
 
+    auto get_repacking_loop_idces = [](const snippets::lowered::ExpressionPtr& parent_expr) {
+        // Repacking may be extracted outside the snippets kernel. In this case, brgemm parent expression is a parameter.
+        if (is_type<ov::op::v0::Parameter>(parent_expr->get_node()))
+            return std::vector<size_t>{};
+
+        OPENVINO_ASSERT(is_type<snippets::lowered::BufferExpression>(parent_expr),
+                        "In case of repacking brgemm expr must have BufferExpression on B input");
+        const auto buffer_parent_ports = parent_expr->get_input_port(0).get_connected_ports();
+        OPENVINO_ASSERT(buffer_parent_ports.size() == 1,
+                        "Parent of brgemm repacking buffer must be connected only to the buffer");
```

Comment:
May be do you mean `get_input_count()` instead of `get_input_port(0).get_connected_ports()`?
First one returns count of input ports - they might be various.
Second one return all connected parent ports connected to the input - it always one source port for any expression input.

By the way, If you want to get parent, you can get it using `parent_expr->get_input_port_connector(0)->get_source().get_expr()` and then you do not need to get `std::set` of connected ports and use iterator `begin` there

Reply:
Thanks for the suggestion, this code was really unclear. I realized that we can use `brgemm_utils::repacking::get_copy_b_expr` here, so now it's more understandable I hope, please take a look